restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/NumberSelectorService.java

```diff
@@ -0,0 +1,245 @@
+import java.util.regex.Pattern;
+import org.apache.log4j.Logger;
+import org.restcomm.connect.commons.dao.Sid;
+import org.restcomm.connect.dao.IncomingPhoneNumbersDao;
+import org.restcomm.connect.dao.entities.IncomingPhoneNumber;
+import org.restcomm.connect.dao.entities.IncomingPhoneNumberFilter;
+
+public class NumberSelectorService {
+
+    private static Logger logger = Logger.getLogger(NumberSelectorService.class);
+
+    private IncomingPhoneNumbersDao numbersDao;
+
+    public NumberSelectorService(IncomingPhoneNumbersDao numbersDao) {
+        this.numbersDao = numbersDao;
+    }
+
+    /**
+     *
+     * @param phone
+     * @return a list of strings to match number based on different formats
+     */
+    private List<String> createPhoneQuery(String phone) {
+        List<String> numberQueries = new ArrayList<String>(10);
+        numberQueries.add(phone);
+        // Format the destination to an E.164 phone number.
+        if (!(phone.contains("*") || phone.contains("#"))) {
+            try {
+                final PhoneNumberUtil phoneNumberUtil = PhoneNumberUtil.getInstance();
+                String usFormatNum = phoneNumberUtil.format(phoneNumberUtil.parse(phone, "US"), PhoneNumberUtil.PhoneNumberFormat.E164);
+                if (!numberQueries.contains(usFormatNum)) {
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("Adding US Format num to queries:" + usFormatNum);
+                    }
+                    numberQueries.add(usFormatNum);
+                }
+            } catch (NumberParseException e) {
+                //logger.error("Exception when try to format : " + e);
+            }
+        }
+        if (phone.startsWith("+")) {
+            //remove the (+) and check if exists
+            String noPlusNum = phone.replaceFirst("\\+", "");
+            if (!numberQueries.contains(noPlusNum)) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Adding No Plus Num:" + noPlusNum);
+                }
+                numberQueries.add(noPlusNum);
+            }
+        } else {
+            String plusNum = "+".concat(phone);
+            if (!numberQueries.contains(plusNum)) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Adding Plus Num:" + plusNum);
+                }
+                numberQueries.add(plusNum);
+            }
+        }
+        return numberQueries;
+    }
+
+    private IncomingPhoneNumber findSingleNumber(String number,
+            Sid sourceOrganizationSid, Sid destinationOrganizationSid) {
+        IncomingPhoneNumber matchedNumber = null;
+        IncomingPhoneNumberFilter.Builder filterBuilder = IncomingPhoneNumberFilter.Builder.builder();
+        filterBuilder.byPhoneNumber(number);
+        if (destinationOrganizationSid != null) {
+            filterBuilder.byOrgSid(destinationOrganizationSid.toString());
+        }
+        if (sourceOrganizationSid!= null &&
```

Comment:
@jaimecasero can you elaborate on the reason behind this rule?
Is it?
Client XYZ the belongs to Org A dial to DID +123456789 that belongs to Org B? and thus since we don't allow calling sip numbers between Organizations the number to be matched must NOT BE pureSip ?

Reply:
which rule 
-if destOrg != null -> byDestOrg
-if srcOrg !0null && srcOrg!=destOrg -> bypureSIP(false)

i think this use case is supported by next rule srcOrg!=null and srcOrg!=destOrg -> byPureSIP(false) so sipnumbers are not included in results...

Notice SMPPHandler will not provide any srcOrg or Destorg, but we still need to apply some numberSlection logic to SMPP...

Reply:
@jaimecasero I refer to the rule that the comment is attached, which is 
```
if (sourceOrganizationSid!= null &&
                !sourceOrganizationSid.equals(destinationOrganizationSid)) {
            filterBuilder.byPureSIP(Boolean.FALSE);
        }
```

I think there is a misunderstanding here, maybe I didn't completely understand the role of NumberSelectorService.findSingleNumber() 

As far as I can understand this method is used to match a specific number only, right?

I think we absolutely need more comments and details on the NumberSelectorService

Reply:
this particular rule was copied from previous OrganizationUtil. I can provide more comments though..